chainindex/ddls.go

```diff
@@ -0,0 +1,93 @@
+package chainindex
+
+const DefaultDbFilename = "chainindex.db"
+
+const (
+	stmtGetNonRevertedMessageInfo = "SELECT tipset_key_cid, height FROM tipset_message WHERE message_cid = ? AND reverted = 0"
+	stmtGetMsgCidFromEthHash      = "SELECT message_cid FROM eth_tx_hash WHERE tx_hash = ?"
+	stmtInsertEthTxHash           = "INSERT INTO eth_tx_hash (tx_hash, message_cid) VALUES (?, ?) ON CONFLICT (tx_hash) DO UPDATE SET inserted_at = CURRENT_TIMESTAMP"
+
+	stmtInsertTipsetMessage = "INSERT INTO tipset_message (tipset_key_cid, height, reverted, message_cid, message_index) VALUES (?, ?, ?, ?, ?) ON CONFLICT (tipset_key_cid, message_cid) DO UPDATE SET reverted = 0"
+
+	stmtHasTipset                 = "SELECT EXISTS(SELECT 1 FROM tipset_message WHERE tipset_key_cid = ?)"
+	stmtUpdateTipsetToNonReverted = "UPDATE tipset_message SET reverted = 0 WHERE tipset_key_cid = ?"
+
+	stmtUpdateTipsetToReverted = "UPDATE tipset_message SET reverted = 1 WHERE tipset_key_cid = ?"
+
+	stmtRemoveTipsetsBeforeHeight = "DELETE FROM tipset_message WHERE height < ?"
+
+	stmtRemoveEthHashesOlderThan = `DELETE FROM eth_tx_hash WHERE inserted_at < datetime('now', ?);`
+
+	stmtUpdateTipsetsToRevertedFromHeight = "UPDATE tipset_message SET reverted = 1 WHERE height >= ?"
+
+	stmtUpdateEventsToRevertedFromHeight = "UPDATE event SET reverted = 1 WHERE message_id IN (SELECT message_id FROM tipset_message WHERE height >= ?)"
+
+	stmtIsTipsetMessageNonEmpty = "SELECT EXISTS(SELECT 1 FROM tipset_message LIMIT 1)"
+
+	stmtGetMinNonRevertedHeight = `SELECT MIN(height) FROM tipset_message WHERE reverted = 0`
+
+	stmtHasNonRevertedTipset = `SELECT EXISTS(SELECT 1 FROM tipset_message WHERE tipset_key_cid = ? AND reverted = 0)`
+
+	stmtUpdateEventsToReverted = `UPDATE event SET reverted = 1 WHERE message_id IN (
+			SELECT message_id FROM tipset_message WHERE tipset_key_cid = ?
+		)`
+
+	stmtUpdateEventsToNonReverted = `UPDATE event SET reverted = 0 WHERE message_id IN (
+		SELECT message_id FROM tipset_message WHERE tipset_key_cid = ?
+	)`
+
+	stmtGetMsgIdForMsgCidAndTipset = `SELECT message_id FROM tipset_message WHERE tipset_key_cid = ? AND message_cid = ?AND reverted = 0`
+
+	stmtInsertEvent      = "INSERT INTO event (message_id, event_index, emitter_addr, reverted) VALUES (?, ?, ?, ?)"
+	stmtInsertEventEntry = "INSERT INTO event_entry (event_id, indexed, flags, key, codec, value) VALUES (?, ?, ?, ?, ?, ?)"
+)
+
+var ddls = []string{
+	`CREATE TABLE IF NOT EXISTS tipset_message (
+		message_id INTEGER PRIMARY KEY,
+		tipset_key_cid BLOB NOT NULL,
+		height INTEGER NOT NULL,
+		reverted INTEGER NOT NULL,
+		message_cid BLOB,
```

Comment:
any reason to not do this to it?

```suggestion
		message_cid BLOB NOT NULL,
```

Reply:
@rvagg for empty tipsets. Empty tipsets have a cid and exist in the chain store but have no messages or events.